elyra/pipeline/tests/test_kfp_processor.py

```diff
@@ -112,3 +112,22 @@ def test_get_dependency_archive_name(processor):
     filename = processor._get_dependency_archive_name(test_operation)
 
     assert filename == correct_filename
+
+
+def test_collect_envs(processor):
+    pipelines_test_file = 'elyra/pipeline/tests/resources/archive/test.ipynb'
+
+    test_operation = Op(id='this-is-a-test-id',
+                        type='execution-node',
+                        classifier='kfp',
+                        name='test',
+                        filename=pipelines_test_file,
+                        runtime_image='tensorflow/tensorflow:latest')
+
+    envs = processor._collect_envs(test_operation, cos_secret=None, cos_username='Alice', cos_password='secret')
```

Comment:
This test should also cover the `cos_secret` code path, which would not return `envs['AWS_ACCESS_KEY_ID']` and `envs['AWS_SECRET_ACCESS_KEY']` for null/empty.